src/services/services.ts

```diff
@@ -2915,6 +3000,69 @@ namespace ts {
 
     export function createLanguageService(host: LanguageServiceHost,
         documentRegistry: DocumentRegistry = createDocumentRegistry(host.useCaseSensitiveFileNames && host.useCaseSensitiveFileNames(), host.getCurrentDirectory())): LanguageService {
+            const baseService = createUnextendedLanguageService(host, documentRegistry);
+            const extensions = baseService.getProgram().getCompilerExtensions()["language-service"];
+            const instantiatedExtensions = map(extensions, extension => new extension.ctor(ts, host, baseService, documentRegistry, extension.args));
+
+            function wrap(key: string): Function {
+                return (...args: any[]) => {
+                    if (instantiatedExtensions && instantiatedExtensions.length) {

```

Comment:
While I can (and am) moving the extension count outside the block, since this is all really backed by JS, it's wholly possible for the members on the imported classes to change as they are executing. I seriously hope nobody ever does that, but considering the possibility makes this code more defensive.